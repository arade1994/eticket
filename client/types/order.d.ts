import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: Date;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
