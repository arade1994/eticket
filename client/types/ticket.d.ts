export interface Ticket {
  id: string;
  title: string;
  price: number;
  userId: string;
  category: string;
  version: number;
}
