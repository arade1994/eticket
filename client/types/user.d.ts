import { type IncomingMessage } from "http";

export interface User {
  id: string;
  firstName: string;
  lastName: string;
  age: number;
  email: string;
  password: string;
}

export interface Rating {
  id: string;
  comment: string;
  rate: number;
  userId: string;
  ratedUserId: string;
}

export interface RequestWithUser extends IncomingMessage {
  currentUser?: { id: string; email: string };
}
