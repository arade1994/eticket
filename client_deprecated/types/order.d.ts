import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: string;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
