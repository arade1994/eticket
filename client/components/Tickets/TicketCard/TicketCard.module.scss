.ticketCard {
  width: 100%;
  height: 300px;
  box-shadow: 1px 0px 6px gray;
  border-radius: 5px;
  overflow: hidden;
  padding: 10px;
  cursor: pointer;
  text-decoration: none;

  & .coverPhoto {
    height: 150px;
  }

  & .ticketTitle {
    margin: 0;
    font-weight: bold;
    color: #004155;
    opacity: 0.7;
  }

  & .ticketPrice {
    display: flex;
    justify-content: space-between;
    gap: 5px;
    margin-top: 15px;
    opacity: 0.7;

    & p {
      font-weight: bold;
      color: #004155;
      margin: 0;
    }
  }

  & .ticketSeller {
    display: flex;
    justify-content: space-between;
    gap: 5px;
    opacity: 0.7;

    & p {
      margin-top: 5px;
      color: #004155;
      font-weight: bold;
    }
  }

  & .ticketCategory {
    justify-self: center;
    font-weight: bold;
    color: #004155;
    opacity: 0.7;
  }
}
