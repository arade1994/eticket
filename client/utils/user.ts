import { Ticket } from "../types/ticket";
import { User } from "../types/user";

export const getNumOfCreatedTickets = (user: User, tickets: Ticket[]) =>
  tickets?.filter((ticket) => ticket.userId === user.id)?.length;
