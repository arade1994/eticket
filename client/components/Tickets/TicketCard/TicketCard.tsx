import { useMemo } from "react";
import Image from "next/image";
import Link from "next/link";
import { FaStar } from "react-icons/fa";

import coverImage from "../../../assets/images/cover.png";
import { type Ticket } from "../../../types/ticket";
import { type Rating, type User } from "../../../types/user";
import { getUserRating } from "../../../utils/users";

import classes from "./TicketCard.module.scss";

interface Props {
  ticket: Ticket;
  user?: User;
  ratings: Rating[];
}

const TicketCard: React.FC<React.PropsWithChildren<Props>> = ({
  ticket,
  user,
  ratings,
}) => {
  const rating = useMemo(
    () => getUserRating(ticket.userId, ratings),
    [ratings, ticket.userId]
  );

  return (
    <Link className={classes.ticketCard} href={`/tickets/${ticket.id}`}>
      <Image
        alt="Ticket cover photo"
        className={classes.coverPhoto}
        height={200}
        src={coverImage}
        width={300}
      />
      <div className={classes.ticketContent}>
        <h4 className={classes.ticketTitle}>{ticket.title}</h4>
        <div className={classes.ticketPrice}>
          <p>${ticket.price}</p>
        </div>
        <div className={classes.ticketSeller}>
          <p>
            Seller: {user ? `${user.firstName} ${user.lastName}` : "Unknown"}
          </p>
          {rating !== "-" && (
            <span className={classes.rating}>
              <FaStar /> {rating}
            </span>
          )}
        </div>
        <span className={classes.ticketCategory}>{ticket.category}</span>
      </div>
    </Link>
  );
};

export default TicketCard;
