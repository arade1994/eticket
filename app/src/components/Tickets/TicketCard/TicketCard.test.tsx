import { describe, expect, test } from "vitest";

import { render } from "@testing-library/react";

import mockDb from "../../../../../client_deprecated/mock/db.json";
import { type Ticket } from "../../../types/ticket";
import { type Rating, type User } from "../../../types/user";

import TicketCard from "./TicketCard";

const mockTicket = mockDb.tickets[0] as unknown as Ticket;
const mockUser = mockDb.users.find(
  (user) => user.id === mockTicket.userId
) as unknown as User;
const mockRatings = mockDb.ratings as unknown as Rating[];

describe("<TicketCard />", () => {
  test("it should match snapshot when rendered by default", () => {
    const { baseElement } = render(
      <TicketCard ratings={mockRatings} ticket={mockTicket} user={mockUser} />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
