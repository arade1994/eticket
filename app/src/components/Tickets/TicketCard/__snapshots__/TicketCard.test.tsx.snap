// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TicketCard /> > it should match snapshot when rendered by default 1`] = `
<body>
  <div>
    <a
      class="_ticketCard_e6793d"
      href="/tickets/t1"
    >
      <img
        alt="Ticket cover photo"
        class="_coverPhoto_e6793d"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
        srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="_ticketContent_e6793d"
      >
        <h4
          class="_ticketTitle_e6793d"
        >
          NBA Finals Game 7
        </h4>
        <div
          class="_ticketPrice_e6793d"
        >
          <p>
            $
            450.00
          </p>
        </div>
        <div
          class="_ticketSeller_e6793d"
        >
          <p>
            Seller: 
            Ante Rade
          </p>
        </div>
        <span
          class="_ticketCategory_e6793d"
        >
          sport game
        </span>
      </div>
    </a>
  </div>
</body>
`;
